### PR TITLE
Add more tests for strings with unicode (escapes)

### PR DESCRIPTION
**Pull request by axic**
Note: we have comprehensive tests for UTF-8 in `test/libsolutil`, but we did not have too much through the solidity parser.

### CODE REVIEW

File: test/libsolidity/semanticTests/strings/unicode_string.sol [hunk 1]
```diff
@@ -0,0 +1,7 @@
+contract C {
+    function f() public pure returns (string memory) {
+        return "😃, 😭, and 😈";
```
Comment (chriseth) [inline test/libsolidity/semanticTests/strings/unicode_string.sol:3]: 😃

**Review by chriseth — APPROVED** [review]
